reconcile/utils/terrascript_aws_client.py

```diff
@@ -870,6 +871,22 @@ def get_provider_alias(account: Mapping[str, str]) -> str:
         role_name = awsh.get_id_from_arn(account["assume_role"])
         return f"account-{uid}-{role_name}"
 
+    @staticmethod
+    def get_resource_lifecycle(
+        common_values: dict[str, Any],
+    ) -> Optional[dict[str, Any]]:
+        if lifecycle := common_values.get("lifecycle"):
```

Comment:
would it make sense to use the generated type here to gain some type safety? Like

```
lifecycle := NamespaceTerraformResourceLifecycleV1(common_values.get("lifecycle")**)
...
return lifecycle.dict(by_alias=True)
```

Reply:
https://github.com/app-sre/qontract-reconcile/pull/4042/commits/471e476a878f168da8ee66ba0556db569a105bb7